Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/ui/details/PersonDetailsFragment.kt

```diff
@@ -84,6 +88,7 @@ class PersonDetailsFragment : Fragment(R.layout.person_details_fragment), AutoIn
                 )
             }
         }
+        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner) { viewModel.dismissAdmissionStateBadge() }
```

Comment:
I think you can do this in `onStop` and it will cover both cases `<- and back`

Reply:
I tried it before implementing it this way but it wasn't working as expected. It gets dismissed by actually opening a certificate details and then going back to the person details screen, but it's not working when just going back from the person details screen to the certificates overview screen.